test/server/guarddog_impl_test.cc

```diff
@@ -174,6 +206,38 @@ TEST_P(GuardDogAlmostDeadTest, MultiKillNoFinalCheckTest) {
   SetupForMultiDeath();
 }
 
+TEST_P(GuardDogDeathTest, MultiKillThresholdDeathTest) {
+  auto die_function = [&]() -> void {
+    setupForMultiDeathThreshold();
+
+    // Pet the last two dogs so we're just at the threshold that causes death.
+    dogs_.at(4)->touch();
+    dogs_.at(3)->touch();
+
+    time_system_->advanceTimeWait(std::chrono::milliseconds(2)); // 1 ms past multi-death.
+    guard_dog_->forceCheckForTest();
+  };
+  EXPECT_DEATH(die_function(), "");
+}
+
+TEST_P(GuardDogAlmostDeadTest, MultiKillUnderThreshold) {
+  // This does everything the death test does except it pets an additional watchdog
+  // that causes us to be under the threshold (60%) of multikill death.
+  auto die_function = [&]() -> void {
+    setupForMultiDeathThreshold();
+
+    // Pet the last three dogs so we're just under the threshold that causes death.
+    dogs_.at(4)->touch();
+    dogs_.at(3)->touch();
+    dogs_.at(2)->touch();
+
+    time_system_->advanceTimeWait(std::chrono::milliseconds(2)); // 1 ms past multi-death.
```

Comment:
You may want to add a second advance after this code block to verify that having 4 watchdogs with stale pettings will result in a kill.

Reply:
Think the other death test >= threshold should handle that case. It's also more consistent with the rest of the tests in the file.